### PR TITLE
boards: arm: nrf5340dk_nrf5340: move defaults to Kconfig.defconfig

### PR DESCRIPTION
**Pull request by gmarull**
Kconfig.defconfig is the file meant for adjusting other Kconfig options
defaults. Some other minor adjustments have been made.

### CODE REVIEW

**Review by carlescufi — DISMISSED** [review]

File: boards/arm/nrf5340dk_nrf5340/Kconfig [hunk 1]
```diff
@@ -95,22 +46,11 @@ config DOMAIN_CPUNET_BOARD
...
-	default y if BT
-
-config BT_ECC
-	default y if BT
-
 config DOMAIN_CPUAPP_BOARD
-	string
+	string "Board used by the CPUAPP domain"
 	default "nrf5340dk_nrf5340_cpuapp" if BOARD_NRF5340DK_NRF5340_CPUNET
+	depends on BOARD_NRF5340DK_NRF5340_CPUNET
```
Comment (anangl) [inline boards/arm/nrf5340dk_nrf5340/Kconfig:52]: The condition for the default value is redundant.
```suggestion
	default "nrf5340dk_nrf5340_cpuapp"
	depends on BOARD_NRF5340DK_NRF5340_CPUNET
```

File: boards/arm/nrf5340dk_nrf5340/Kconfig [hunk 2]
```diff
@@ -84,7 +35,7 @@ config BOARD_ENABLE_CPUNET
 	default y if (BT || NRF_802154_SER_HOST)
 
 config DOMAIN_CPUNET_BOARD
-	string
+	string "Board used by the CPUNET domain"
 	default "nrf5340dk_nrf5340_cpunet" if BOARD_NRF5340DK_NRF5340_CPUAPP || BOARD_NRF5340DK_NRF5340_CPUAPP_NS
```
Comment (anangl) [inline boards/arm/nrf5340dk_nrf5340/Kconfig:39]: Same here. The whole option is defined under `if BOARD_NRF5340DK_NRF5340_CPUAPP || BOARD_NRF5340DK_NRF5340_CPUAPP_NS`.

File: boards/arm/nrf5340dk_nrf5340/Kconfig [hunk 3]
```diff
@@ -84,7 +35,7 @@ config BOARD_ENABLE_CPUNET
 	default y if (BT || NRF_802154_SER_HOST)
 
 config DOMAIN_CPUNET_BOARD
-	string
+	string "Board used by the CPUNET domain"
```
[the remark this answers is not in the document]
Comment (gmarull) [reply boards/arm/nrf5340dk_nrf5340/Kconfig]: removed

File: boards/arm/nrf5340dk_nrf5340/Kconfig (hunk 1 above)
Comment (gmarull) [reply boards/arm/nrf5340dk_nrf5340/Kconfig]: removed, good catch

File: boards/arm/nrf5340dk_nrf5340/Kconfig (hunk 2 above)
Comment (gmarull) [reply boards/arm/nrf5340dk_nrf5340/Kconfig]: removed

**Review by anangl — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**